lib/client/api.go

```diff
@@ -326,7 +326,9 @@ func (tc *TeleportClient) SSH(command []string, runLocally bool) error {
 }
 
 // Join connects to the existing/active SSH session
-func (tc *TeleportClient) Join(sessionID session.ID) (err error) {
+func (tc *TeleportClient) Join(sid string) (err error) {
```

Comment:
it's a re-typed string, causing a ton of typecasts everywhere else. so you're not gaining anything except making caller's life continuously difficult.